kotlin/display-subtype-feature-layer/src/main/java/com/esri/arcgisruntime/sample/displaysubtypefeaturelayer/MainActivity.kt

```diff
@@ -0,0 +1,112 @@
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.layers.SubtypeFeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.sublayer_control_layout.*
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // setup map with basemap and an initial viewpoint
+    mapView.map = ArcGISMap(Basemap.createStreetsNightVector()).apply {
+      initialViewpoint = Viewpoint(
+        Envelope(
+          -9812691.11079696,
+          5128687.20710657,
+          -9812377.9447607,
+          5128865.36767282,
+          SpatialReferences.getWebMercator()
+        )
+      )
+    }
+
+    // on any navigation on the map view
+    mapView.addMapScaleChangedListener {
+      currentMapScaleTextView.text =
+        getString(R.string.current_map_scale_text, mapView.mapScale.roundToInt())
+    }
+
+    // create a subtype feature layer from a service feature table
+    val subtypeFeatureLayer =
+      SubtypeFeatureLayer(ServiceFeatureTable("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100"))
+    // add it to the map
+    mapView.map.operationalLayers.add(subtypeFeatureLayer)
+
+    // once the subtype feature layer is loaded
+    subtypeFeatureLayer.addDoneLoadingListener {
+      // create a subtype sublayer
+      val subtypeSublayer = subtypeFeatureLayer.getSublayerWithSubtypeName("Street Light").apply {
+        isLabelsEnabled = true
+        labelDefinitions.add(LabelDefinition.fromJson(getString(R.string.label_json)))
+      }
+
+      // show subtype sublayer when checked, hide when unchecked
+      showSubtypeSublayerCheckBox.setOnClickListener {
+        subtypeSublayer.isVisible = showSubtypeSublayerCheckBox.isChecked
+      }
+
+      // get the original renderer of the subtype sublayer
+      val originalRenderer = subtypeSublayer.renderer
+
+      // when the selected radio button changes
+      rendererRadioGroup.setOnCheckedChangeListener { _, checkedId ->
+        // set the sublayer renderer to
+        subtypeSublayer.renderer = when (checkedId) {
+          alternativeRendererButton.id -> {
+            // use an alternative renderer
+            SimpleRenderer(
+              SimpleMarkerSymbol(SimpleMarkerSymbol.Style.DIAMOND, Color.MAGENTA, 20f)
+            )
+          }
+          originalRendererButton.id -> {
+            // use the original renderer
+            originalRenderer
+          }
+          else -> {
+            error("Invalid radio button.")
```

Comment:
just wondering, would we ever get this error in the sample?

Reply:
No, but Kotlin complains if a when clause isn't "exhaustive".